frontend/test/metabase/scenarios/sharing/public.cy.spec.js

```diff
@@ -201,4 +201,130 @@ describe("scenarios > public", () => {
       }),
     );
   });
+
+  describe("public dashboard with parameters linked to a custom field (metabase#25473)", () => {
+    it("should show text parameter in public dashboards", () => {
+      // Setup
+      const customField = {
+        name: "email_name",
+        // Custom field is `concat([Email], [Name])`
+        definition: [
+          "concat",
+          ["field", PEOPLE.EMAIL, null],
+          ["field", PEOPLE.NAME, null],
+        ],
+      };
+      const questionDetails = {
+        name: "Question",
+        query: {
+          "source-table": PEOPLE_ID,
+          expressions: {
+            [customField.name]: customField.definition,
+          },
+        },
+      };
+      const textContainsFilter = {
+        id: "890abcde",
+        name: "Text contains",
+        slug: "text_contains",
+        type: "string/contains",
+      };
+      const dashboardDetails = {
+        parameters: [textContainsFilter],
+      };
+
+      cy.createQuestionAndDashboard({
+        questionDetails,
+        dashboardDetails,
+      }).then(({ body: { id, card_id, dashboard_id } }) => {
+        cy.wrap(dashboard_id).as("dashboardId");
+
+        mapParameters({
+          id,
+          card_id,
+          dashboard_id,
+          parameter: textContainsFilter,
+          customFieldName: customField.name,
+        });
+      });
+
+      cy.get("@dashboardId").then(dashboardId => {
+        visitDashboard(dashboardId);
+      });
+
+      // Enable public sharing
+      cy.icon("share").click();
+
+      cy.findByText("Enable sharing")
+        .parent()
+        .find("input[type=checkbox]")
+        .check();
+
+      cy.contains("Public link")
+        .parent()
+        .find("input")
+        .then($input => {
+          const dashboardPublicLink = $input[0].value;
+          cy.visit(dashboardPublicLink);
+        });
+
+      // Assert public dashboard
```

Comment:
Given the custom field as `concat([Email], [Name])`
There are 3 assertions here.
1. search by email
2. search by name
3. search by email+name

Each with different person, so we know it's working.